### PR TITLE
Added a problem and solution to the Dijkstra module.

### PR DESCRIPTION
**Pull request by jessechoe10**

### CODE REVIEW

File: solutions/cf-20C.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#define all(x) begin(x), end(x)
+
+int main() {
+	priority_queue<pi, vector<pi>, greater<pi>> pq;
+	int n, m; cin >> n >> m;
+	for(int i = 0; i < m; i++){
+		int a, b, w; cin >> a >> b >> w;
+		a--, b--; adj[a].push_back({w, b}), adj[b].push_back({w, a});
+	}
+	vector<ll> dist(n, INF); dist[0] = 0;
```
Comment (maggie-j-liu) [inline solutions/cf-20C.mdx:34]: did you test the code?? what's `INF`?

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/cf-20C.mdx (hunk 1 above)
Comment (jessechoe10) [reply solutions/cf-20C.mdx:38]: shit mb forgot to add this after removing my template.

File: content/4_Gold/SP.problems.json [hunk 2]
```diff
@@ -16,6 +16,18 @@
     }
   ],
   "dijk": [
+    {
```
[the remark this answers is not in the document]
Comment (jessechoe10) [reply content/4_Gold/SP.problems.json:19]: Ohhhhhhh, that's why u closed the prob suggestions lol. Alright, that's fine haha

[the remark this answers is not in the document]
Comment (jessechoe10) [reply content/4_Gold/SP.problems.json:19]: Oh wait, can we add this to Additional Practice for Gold?

[the remark this answers is not in the document]
Comment (maggie-j-liu) [reply content/4_Gold/SP.problems.json:19]: idk, ask someone from core? 👀

File: content/4_Gold/SP.problems.json [hunk 3]
```diff
@@ -16,6 +16,18 @@
...
+      "uniqueId": "cf-20C",
```
Comment (dongliu0426) [inline content/4_Gold/SP.problems.json:20]: I'm against adding this problem to this module/additional practice because it's a standard dijkstra problem.

Comment (dongliu0426) [reply content/4_Gold/SP.problems.json:20]: It repeats the idea of https://cses.fi/problemset/task/1671 (focus problem from the module).

Comment (jessechoe10) [reply content/4_Gold/SP.problems.json:20]: > It repeats the idea of https://cses.fi/problemset/task/1671 (focus problem from the module).

What about replacing https://cses.fi/problemset/task/1671 with this problem?

Comment (dongliu0426) [reply content/4_Gold/SP.problems.json:20]: why?

Comment (dongliu0426) [reply content/4_Gold/SP.problems.json:20]: why?

Comment (jessechoe10) [reply content/4_Gold/SP.problems.json:20]: > why?

They're basically the same problem, except https://cses.fi/problemset/task/1671 doesn't have finding the shortest path. I feel like finding the shortest path teaches you more idk.

Comment (jessechoe10) [reply content/4_Gold/SP.problems.json:20]: > why?

Did you repeat "why?" on purpose btw?

Comment (dongliu0426) [reply content/4_Gold/SP.problems.json:20]: oops bad internet

Comment (dongliu0426) [reply content/4_Gold/SP.problems.json:20]: > They're basically the same problem, except https://cses.fi/problemset/task/1671 doesn't have finding the shortest path. I feel like finding the shortest path teaches you more idk.

> idk, ask someone from core? 👀

**Review by ryanchou-dev — APPROVED** [review]
nice code and great editorial, great job!

File: content/4_Gold/SP.problems.json (hunk 3 above)
Comment (thecodingwizard) [reply content/4_Gold/SP.problems.json:20]: In general I like CSES better than CF as focus problem. But I didn't actually read the problems 😆

File: solutions/cf-20C.mdx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+#define s second
+#define all(x) begin(x), end(x)
+
+vector<pi> adj[100000];
+ 
+const ll INF = 1e11;
+	
+int main() {
+	priority_queue<pi, vector<pi>, greater<pi>> pq;
+	int n, m; cin >> n >> m;
```
Comment (SansPapyrus683) [inline solutions/cf-20C.mdx:33]: no one letter variable names?

File: solutions/cf-20C.mdx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	for(int i = 0; i < m; i++){
+		int a, b, w; cin >> a >> b >> w;
+		a--, b--; adj[a].push_back({w, b}), adj[b].push_back({w, a});
```
Comment (SansPapyrus683) [inline solutions/cf-20C.mdx:36]: can you put the third statement on its own line

File: solutions/cf-20C.mdx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		for(pi vertices: adj[a.s]){
+			if(dist[a.s] + vertices.f < dist[vertices.s]){
+				dist[vertices.s] = dist[a.s] + vertices.f;
+				pq.push({dist[vertices.s], vertices.s});
+			}
+		}
+	}
+	if(dist[n - 1] == INF){
+		cout << -1 << endl;
+	} else {
```
Comment (SansPapyrus683) [inline solutions/cf-20C.mdx:55]: why spaces around the `else` but not around any other control flow

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-20C.mdx (hunk 6 above)
Comment (jessechoe10) [reply solutions/cf-20C.mdx:55]: wait hold up. If this isn't going on the guide, then we don't really have to be strict about this stuff. Do we?

Comment (SansPapyrus683) [reply solutions/cf-20C.mdx:55]: if this isn't going on the guide why is it in a pr on usaco-guide

Comment (jessechoe10) [reply solutions/cf-20C.mdx:55]: [Perhaps for this reason](https://github.com/cpinitiative/usaco-guide/pull/1959#discussion_r672660506)

Comment (SansPapyrus683) [reply solutions/cf-20C.mdx:55]: ooooooohhhh ok
uh still it's good to like make the code look good i mean
but yeah i won't lose my crap about this

Comment (jessechoe10) [reply solutions/cf-20C.mdx:55]: > uh still it's good to like make the code look good i mean
tbf there are a lot of solutions with shittier code than this so like i think we're fine idk.

Comment (maggie-j-liu) [reply solutions/cf-20C.mdx:55]: 👀 😞 that's not a reason to not format your code nicely...

**Comment by maggie-j-liu** [comment]
add the problem extraProblems.json to fix the build

**Comment by dolphingarlic** [comment]
Do we really need to add a problem that's just "implement Dijkstra's algorithm"? :/

**Comment by jessechoe10** [comment]
> Do we really need to add a problem that's just "implement Dijkstra's algorithm"? :/

We aren't adding it lmao. It'll just be under `extraProblems.json`.

**Review by darren-yao — APPROVED** [review]

**Pull request merged**